File: python-sdk/free_endpoints.py
"""
RoutePlex Python SDK — Free Endpoints (no API key charge)

pip install routeplex
"""

import os
from routeplex import RoutePlex

client = RoutePlex(api_key=os.environ["ROUTEPLEX_API_KEY"])

# Cost estimation — free
estimate = client.estimate("Write a detailed blog post about AI in healthcare")
print(f"Model: {estimate.model}")
print(f"Estimated tokens: {estimate.total_tokens}")
print(f"Estimated cost: ${estimate.estimated_cost_usd:.6f}")
print(f"Confidence: {estimate.confidence}")

print()

# Prompt enhancement — free
result = client.enhance("tell me about kubernetes")
if result.changed:
    print(f"Original: tell me about kubernetes")
    print(f"Enhanced: {result.enhanced_prompt}")
    print(f"Type: {result.query_type}")
else:
    print("Prompt was already good!")

print()

# List models — free
models = client.list_models()
for m in models:
    print(f"  {m.id:25s} {m.provider:10s} {m.tier}")

File: node-sdk/free_endpoints.mjs
/**
 * RoutePlex Node.js SDK — Free Endpoints (no API key charge)
 *
 * npm install @routeplex/node
 */

import { RoutePlex } from "@routeplex/node";

const client = new RoutePlex({ apiKey: process.env.ROUTEPLEX_API_KEY });

// Cost estimation — free
const estimate = await client.estimate("Write a detailed blog post about AI in healthcare");
console.log(`Model: ${estimate.model}`);
console.log(`Estimated tokens: ${estimate.totalTokens}`);
console.log(`Estimated cost: $${estimate.estimatedCostUsd.toFixed(6)}`);
console.log(`Confidence: ${estimate.confidence}`);

console.log();

// Prompt enhancement — free
const result = await client.enhance("tell me about kubernetes");
if (result.changed) {
  console.log("Original: tell me about kubernetes");
  console.log(`Enhanced: ${result.enhancedPrompt}`);
  console.log(`Type: ${result.queryType}`);
} else {
  console.log("Prompt was already good!");
}

console.log();

// List models — free
const models = await client.listModels();
for (const m of models) {
  console.log(`  ${m.id.padEnd(25)} ${m.provider.padEnd(10)} ${m.tier}`);
}
